engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_week_stream_logs_strategy.go

```diff
@@ -162,8 +162,20 @@ func (strategy *PerWeekStreamLogsStrategy) getRetainedLogsFilePaths(
 	enclaveUuid, serviceUuid string) []string {
 	var paths []string
 
-	// get log file paths as far back as they exist
+	// scan for first existing log file
+	firstWeekWithLogs := 0
 	for i := 0; i < (retentionPeriodInWeeks + 1); i++ {
+		year, week := strategy.time.Now().Add(time.Duration(-i) * oneWeek).ISOWeek()
```

Comment:
@tedim52 is it possible this can behave unexpectedly during winter/summer time transitions, leap years etc? May not be problematic but potentially worth making a function for this and running a few unit tests to make sure it works as you expect.

Reply:
Good callout. This line primarily depends on the `year` and `week` returned by [`ISOWeek()`](https://pkg.go.dev/time#Time.ISOWeek). `ISOWeek` will handle decrementing weeks across a year correctly, `(year, 1) -> (year-1, 52)` which I've got a unit test for. The only interesting things that can happen here are that some years have either 52 or 53 weeks, but `ISOWeek` will still handle this transition. I'll add a test for the 53 week case just to be sure!